utils/python/CIME/create_test.py

```diff
@@ -378,32 +392,36 @@ def _nlcomp_phase(self, test_name):
         return True
 
     ###########################################################################
-    def _build_phase(self, test_name):
+    def _build_phase(self, test):
     ###########################################################################
+        test_name = test["name"]
         test_dir = self._get_test_dir(test_name)
-        return self._run_phase_command(test_name, "./case.test_build", BUILD_PHASE, from_dir=test_dir)
+        return self._run_phase_command(test, "./case.test_build", BUILD_PHASE, from_dir=test_dir)
 
     ###########################################################################
-    def _run_phase(self, test_name):
+    def _run_phase(self, test):
     ###########################################################################
-        test_dir = self._get_test_dir(test_name)
-        return self._run_phase_command(test_name, "./case.submit", RUN_PHASE, from_dir=test_dir)
+        test_dir = self._get_test_dir(test["name"])
```

Comment:
The testlist.xml includes an optional estimated wallclock time for each test,  this is where we apply it.  And this was the motivation to make each test data structure a dict.   Changing the status stuff seemed like an obvious additional cleanup.